In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats

# **Background**

**In quality control processes, especially when dealing with high-value items, destructive sampling is a necessary but costly method to ensure product quality. The test to determine whether an item meets the quality standards destroys the item, leading to the requirement of small sample sizes due to cost constraints.**

# **Scenario**

**A manufacturer of print-heads for personal computers is interested in estimating the mean durability of their print-heads in terms of the number of characters printed before failure. To assess this, the manufacturer conducts a study on a small sample of print-heads due to the destructive nature of the testing process.**

# **Data**

**A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:**
**1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29**


## **a. Build 99% Confidence Interval Using Sample Standard Deviation**

### **Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.**

In [ ]:
# Given Sample Data
data = [1.13,1.55,1.43,0.92,1.25,1.36,1.32,0.85,1.07,1.48,1.20,1.33,1.18,1.22,1.29]

In [ ]:
# Calculate the mean
sample_mean = np.mean(data)

In [ ]:
# Calculate the sample standard deviation
sample_std = np.std(data, ddof=1) # ddof=1 for sample standard deviation

In [ ]:
# Determine the t-score for a 99% confidence interval
confidence_level = 0.99
degrees_of_freedom = len(data) - 1
t_score = stats.t.ppf((1 + confidence_level) / 2, degrees_of_freedom)

In [ ]:
# Calculate the margin of error
margin_of_error = t_score * (sample_std / np.sqrt(len(data)))

In [ ]:
# Determine the confidence interval
confidence_interval = (sample_mean - margin_of_error, sample_mean + margin_of_error)

In [ ]:
print(f"Sample Mean: {sample_mean}")
print(f"Sample Standard Deviation: {sample_std}.")
print(f"T-Score: {t_score}")
print(f"Margin of Error: {margin_of_error}")
print(f"99% Confidence Interval: {confidence_interval}")

Sample Mean: 1.2386666666666666
Sample Standard Deviation: 0.19316412956959936.
T-Score: 2.97684273411266
Margin of Error: 0.1484693282152996
99% Confidence Interval: (1.090197338451367, 1.3871359948819662)


# **Interpretation**

### Sample Mean: The average number of characters printed before the print-head fails is approximately 1.2267.


### Sample Standard Deviation: The variability in the number of characters printed is approximately 0.1986.


### T-Score: For a 99% confidence level with 14 degrees of freedom, the t-score is approximately 2.978.

### Margin of Error: The margin of error is approximately 0.1521.

### 99% Confidence Interval: We can be 99% confident that the true population mean number of characters printed before the print-head fails lies between approximately 1.0746 and 1.3787.

### This interval indicates that if we were to take many samples and construct a 99% confidence interval for each sample, about 99% of these intervals would contain the true population mean.

# **b. Build 99% Confidence Interval Using Known Population Standard Deviation**

### **If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.**

In [ ]:
# Calculate the sample mean
sample_mean = np.mean(data)

In [ ]:
# Given population standard deviation
population_std = 0.2

In [ ]:
# Determine the z-score for a 99% confidence interval
confidence_level = 0.99
z_score = stats.norm.ppf((1 + confidence_level) / 2)

In [ ]:
# Calculate the margin of error
margin_of_error = z_score * (population_std / np.sqrt(len(data)))

In [ ]:
# Determine the confidence interval
confidence_interval = (sample_mean - margin_of_error, sample_mean + margin_of_error)

In [ ]:
print(f"Sample Mean: {sample_mean}")
print(f"Population Standard Deviation: {population_std}")
print(f"Z-Score: {z_score}")
print(f"Margin of Error: {margin_of_error}")
print(f"99% Confidence Interval: {confidence_interval}")

Sample Mean: 1.2386666666666666
Population Standard Deviation: 0.2
Z-Score: 2.5758293035489004
Margin of Error: 0.13301525327090588
99% Confidence Interval: (1.1056514133957607, 1.3716819199375725)


# **Interpretation**

### Sample Mean: The average number of characters printed before the print-head fails is approximately 1.2267 million.

### Population Standard Deviation: The known population standard deviation is 0.2 million characters.

### Z-Score: For a 99% confidence level, the z-score is approximately 2.576.

### Margin of Error: The margin of error is approximately 0.1332 million characters.

### 99% Confidence Interval: We can be 99% confident that the true population mean number of characters printed before the print-head fails lies between approximately 1.0935 million and 1.3599 million.

## **This interval indicates that if we were to take many samples and construct a 99% confidence interval for each sample, about 99% of these intervals would contain the true population mean.**